kore/src/Kore/Debug.hs

```diff
@@ -547,91 +547,97 @@ instance Debug GHC.SrcLoc
 debugPrecGeneric
     :: forall a ann
     .  (Generic a, HasDatatypeInfo a, All2 Debug (Code a))
-    => Int  -- ^ Surrounding precedence
-    -> a
+    => a
+    -> Int  -- ^ Surrounding precedence
     -> Doc ann
-debugPrecGeneric precOut a =
-    debugPrecSOP precOut (SOP.datatypeInfo $ Proxy @a) (SOP.from a)
+debugPrecGeneric a =
+    debugPrecAux (SOP.datatypeInfo $ Proxy @a) (debugSOP (SOP.from a))
 
-debugPrecSOP
+debugPrecAux
     :: forall xss ann
-    .  All2 Debug xss
-    => Int  -- ^ Surrounding precedence
-    -> DatatypeInfo xss
-    -> SOP I xss
+    .  (All SOP.Top xss)
+    => DatatypeInfo xss
+    -> SOP (K (Int -> Doc ann)) xss
+    -> Int  -- ^ Surrounding precedence
     -> Doc ann
-debugPrecSOP precOut datatypeInfo (SOP sop) =
-    SOP.hcollapse $ SOP.hczipWith pAllDebug debugConstr constrs sop
+debugPrecAux datatypeInfo (SOP sop) =
+    SOP.hcollapse $ SOP.hzipWith debugConstr constrs sop
   where
-    pDebug = Proxy :: Proxy Debug
-    pAllDebug = Proxy :: Proxy (All Debug)
-
     constrs :: NP ConstructorInfo xss
     constrs =
         case datatypeInfo of
             SOP.ADT     _ _ cs -> cs
             SOP.Newtype _ _ c  -> c :* Nil
 
-    precConstr, precRecord :: Int
-    precConstr = 10  -- precedence of function application
-    precRecord = 11  -- precedence of record syntax
+precConstr, precRecord :: Int
+precConstr = 10  -- precedence of function application
+precRecord = 11  -- precedence of record syntax
 
-    debugConstr
-        :: All Debug xs
-        => ConstructorInfo xs  -- ^ Constructor
-        -> NP I xs             -- ^ Arguments
-        -> K (Doc ann) xs
+debugConstr
+    :: ConstructorInfo xs  -- ^ Constructor
+    -> NP (K (Int -> Doc ann)) xs             -- ^ Arguments
+    -> K (Int -> Doc ann) xs
 
-    debugConstr (SOP.Constructor name) args =
-        K $ parens (precOut >= precConstr && (not . null) args')
-        $ Pretty.nest 4
+debugConstr (SOP.Constructor name) args =
+    K $ \precOut ->
+        parens (precOut >= precConstr && (not . null) args')
+        . Pretty.nest 4
         $ Pretty.sep (name' : args')
-      where
-        name' = parens needsParens (Pretty.pretty name)
-          where
-            initial = head name
-            needsParens = (not . Char.isLetter) initial && initial /= '('
-        args' =
-            SOP.hcollapse
-            $ SOP.hcmap pDebug (SOP.mapIK $ debugPrec precConstr) args
-
-    debugConstr (SOP.Infix name _ precInfix) (I x :* I y :* Nil) =
-        K $ parens (precOut >= precInfix)
-        $ Pretty.nest 4
-        $ Pretty.sep
-            [ debugPrec precInfix x
-            , Pretty.pretty name
-            , debugPrec precInfix y
-            ]
-
-    debugConstr (SOP.Record name fields) args =
-        K $ parens (precOut >= precRecord)
-        $ Pretty.align $ Pretty.nest 4 $ Pretty.group $ mconcat
+    where
+    name' = parens needsParens (Pretty.pretty name)
+        where
```

Comment:
Here and in may other places: shouldn't `where` use a two-space indentation

Reply:
Thanks, I had Emacs set up to do that automatically, but I guess it forgot. :slightly_frowning_face: